src/coreclr/jit/promotionliveness.cpp

```diff
@@ -528,31 +499,14 @@ void PromotionLiveness::FillInLiveness()
 
         while (true)
         {
-            GenTree* qmark = nullptr;
-            if (m_compiler->compQmarkUsed)
+            for (GenTree* cur = stmt->GetTreeListEnd(); cur != nullptr; cur = cur->gtPrev)
             {
-                GenTree* dst;
-                qmark = m_compiler->fgGetTopLevelQmark(stmt->GetRootNode(), &dst);
-            }
-
-            if (qmark == nullptr)
-            {
-                for (GenTree* cur = stmt->GetTreeListEnd(); cur != nullptr; cur = cur->gtPrev)
+                // Skip liveness updates/marking for defs; they may be conditionally executed.
+                if ((cur->gtFlags & GTF_VAR_DEF) == 0)
                 {
                     FillInLiveness(life, volatileVars, cur->AsLclVarCommon());
                 }
```

Comment:
This looks like it was updated to the wrong side of the if, this was the code for when there is a QMARK.